### ISSUE TITLE
calamares got this error 'Bad Source source="/run/live/medium/live/filesystem.squashfs" ' 

### ISSUE THREAD

**Issue by hayderctee** [calamares/calamares#1059]
Hi
I'm working on customizing ISO based on PureOS using debootstrap with calamares 3.2.1 and  with xfce desktop environment and it's working fine on liveCD without any problem in version 3.2.1, but when i choose the autologin option during installation process, it's end up at the login screen after rebooting the system, and the file /etc/lightdm/lightdm.conf is still in default options, so i changed the version of calamares to 3.2.2, and now i got this error :
Bad Source source="/run/live/medium/live/filesystem.squashfs"
BTW PureOS is Based on Debian but it's using casper package from ubuntu to set the liveCD.

**Comment by adriaandegroot** [comment]
- LightDM autologin is spotty at best. Your lightdm config file should contain a line containing `autologin-user=` for autologin to be configured (either to the user picked in Calamares, or commented out). Although the DM module has been re-vamped, the actual implementations for the different DMs have not, so you'll need to chase this one yourself. Also, the log file should show you what it's doing.
 - The error message tells you that the source file `/run/live/medium/live/filesystem.squashfs` does not exist. That is something to check on your live CD.

**Comment by highvoltage** [comment]
On casper systems the squashfs is usually in /casper/filesystem.squashfs. Kind of weird that they use that, not sure it's even still maintained. Based on that path you probably got your config from calamare-settings-debian, in that case you can edit /etc/calamares/modules/unpackfs.conf to set the path to your squashfs image.

**Comment by adriaandegroot** [comment]
I will add that Calamares 3.2.2 **does not** install the example config files by default anymore. See the release notes. Debian packages the configs separately. Please review the configuration files you are using.

**Comment by hayderctee** [comment]
> On casper systems the squashfs is usually in /casper/filesystem.squashfs. Kind of weird that they use that, not sure it's even still maintained. Based on that path you probably got your config from calamare-settings-debian, in that case you can edit /etc/calamares/modules/unpackfs.conf to set the path to your squashfs image.

That's right, the file `/etc/calamares/modules/unpackfs.conf` in PureOS vesrion of calamares has the `/cdrom/casper/filesystem.squashfs` as his link to` squashfs` filesystem.
But now i faced a new problem after upgraded calamares version to 3.2.2-1 (i do this by adding debian testing repo and upgraded chroot system), calamares now won't run anymore, and when i tried to run calamares from terminal, i got this error :
`calamares: error while loading shared libraries: libPythonQt-Qt5-Python3.6.so.3: cannot open shared object file: No such file or directory`
after googling this error, some posts say that i should install python3-dev package, so i did this but nothing happens and calamares not running now !

**Comment by hayderctee** [comment]
Now i'm doing some tests on it, and if the error still, i will open a new issue.
thank you so much for helping and replying.

**Comment by adriaandegroot** [comment]
The problem with libPythonQt is one in Debian -- they recompiled Calamares while still linking to a version of libPythonQt that they no longer ship. I guess it's now the Python 3.7 version. Other people have been complaining about this on IRC as well. @highvoltage ?

**Comment by highvoltage** [comment]
Sorry, I missed the notification before. Yes, that's a debian bug in the pythonqt packaging. At least it works in testing again for now, I'm working on a proper fix (See debian bug https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=916317)

**Comment by alvertos00** [comment]
So I might be off subject but I fixed using a work around.... not a hardcore Linux guy but I do have some basic knowledge. So my issue was the same error when I used persistence on my Debian 11 non free USB drive. So first I mounted both of the partitions of my usb stick (a little lsblk and blkid to figure it out), then installed mlocate on my persisted usb drive (did I say that right?) ran an updatedb and then locate filesystem.squashfs. It existed under the 1st partition of my usb drive. So first I tried to copy it into the specific path /run/live/medium/live/ (I had to create the second live folder btw...) but i didn't have enough space so I created a beautifull symlink. Here is the example of the symlink I did while in the folder /run/live/medium/live/ 
ln -s /mnt/sdb1/live/filesystem.squashfs filesystem.squashfs. And that issue was fixed.... now time to fight with good ol'grub!
